pkg/multicluster/razee/razee_cluster_manager.go

```diff
@@ -0,0 +1,241 @@
+	if err != nil {
+		return err
+	}
+	var rCluster types.Cluster
+	if len(razeeClusters) == 0 {
+		err = fmt.Errorf("No clusters found for orgID %v", r.orgId)
+		return err
+	}
+	for _, c := range razeeClusters {
+		// Hack until sat-con library is extended with name field
+		m := c.Metadata.(map[string]interface{})
+		name := fmt.Sprintf("%v", m["name"])
+		if name == cluster {
+			rCluster = c
+		}
+	}
+	if rCluster.ClusterID == "" {
+		err = fmt.Errorf("Cannot find cluster %v", cluster)
+		return err
+	}
+
+	// check group exists
+	groups, err := r.con.Groups.Groups(r.orgId)
+	if err != nil {
+		return err
+	}
+	var group *types.Group
+	var groupUuid string
+	for _, g := range groups {
+		if g.Name == groupName {
+			group = &g
+			groupUuid = g.UUID
+		}
+	}
+	if group == nil {
+		addGroup, err := r.con.Groups.AddGroup(r.orgId, groupName)
+		if err != nil {
+			return err
+		}
+		groupUuid = addGroup.UUID
+	}
+
+	_, err = r.con.Groups.GroupClusters(r.orgId, groupUuid, []string{rCluster.ClusterID})
+	if err != nil {
+		return err
+	}
+
+	// create channel
+	channel, err := r.con.Channels.AddChannel(r.orgId, channelName)
+	if err != nil {
+		return err
+	}
+
+	// create channel version
+	channelVersion, err := r.con.Versions.AddChannelVersion(r.orgId, channel.UUID, version, []byte(content), "")
+	if err != nil {
+		return err
+	}
+
+	// create subscription
+	_, err = r.con.Subscriptions.AddSubscription(r.orgId, channelName, channel.UUID, channelVersion.VersionUUID, []string{groupName})
+	if err != nil {
+		return err
+	}
+
+	r.log.Info("Successfully created subscription! ")
+	return nil
+}
+
+//nolint:golint,unused
+func (r *ClusterManager) UpdateBlueprint(cluster string, blueprint *v1alpha1.Blueprint) error {
+	return nil
+}
+
+//nolint:golint,unused
+func (r *ClusterManager) DeleteBlueprint(cluster string, namespace string, name string) error {
+	channelName := channelName(cluster, namespace, name)
+	channel, err := r.con.Channels.ChannelByName(r.orgId, channelName)
+	if err != nil {
+		return err
+	}
+	for _, s := range channel.Subscriptions {
+		subscription, err := r.con.Subscriptions.RemoveSubscription(r.orgId, s.UUID)
+		if err != nil {
+			return err
+		}
+		if subscription.Success {
+			r.log.Info("Successfully deleted subscription " + subscription.UUID)
+		}
+	}
+	for _, v := range channel.Versions {
+		version, err := r.con.Versions.RemoveChannelVersion(r.orgId, v.UUID)
+		if err != nil {
+			return err
+		}
+		if version.Success {
+			r.log.Info("Successfully deleted version " + version.UUID)
+		}
+	}
+
+	removeChannel, err := r.con.Channels.RemoveChannel(r.orgId, channel.UUID)
+	if err != nil {
+		return err
+	}
+	if removeChannel.Success {
+		r.log.Info("Successfully deleted channel " + removeChannel.UUID)
+	}
+	return nil
+}
+
+//nolint:golint,unused
+func channelName(cluster string, namespace string, name string) string {
+	return "m4d-" + cluster + "-blueprint-" + namespace + "-" + name
```

Comment:
can we swap to `m4d.ibm.com/<cluster>/<namespace>/<name>`? or is it illegal?

Reply:
Nothing is illegal in wonderland:
```
> db.channels.find()
{ "_id" : "a3fe3acb-155f-4fc2-90da-ba17fbb90855", "uuid" : "15b09186-5103-4fbe-bddd-e9ad785e7ebe", "org_id" : "e9077db1-f82f-4cdf-a0e6-06dc7528e0f6", "name" : "m4d.ibm.com/cluster-1/namespace-2/resource-3", "versions" : [ ], "created" : ISODate("2020-12-07T16:20:48.410Z"), "__v" : 0 }
> 
```